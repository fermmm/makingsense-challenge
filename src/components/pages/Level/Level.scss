.mainContainer {
   display: flex;
   flex-direction: column;
   align-items: center;
   justify-content: center;

   .mapContainer {
      display: flex;
      flex-direction: column;
      position: relative;
   }
}
