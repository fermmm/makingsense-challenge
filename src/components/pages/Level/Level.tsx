import { History } from 'history';
import React, { FC } from 'react';
import { hot } from 'react-hot-loader';
import { useHistory } from 'react-router-dom';
// @ts-ignore
import styles from './Level.scss';

interface PropsLevel {
   
}

export const Level: FC<PropsLevel> = () => {
   const history: History = useHistory();

   return (
      <div className={styles.level}>
         not implemented
      </div>
   );
};

export default hot(module)(Level);
