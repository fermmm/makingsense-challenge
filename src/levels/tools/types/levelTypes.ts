export enum TileData {
   Walkable,
   Obstacle,
}

export interface LevelData {
   startPos: Vector;
   finishPos: Vector;
   movesAvailable: number;
   map: TileData[][];
}

export interface Vector {
   x: number;
   y: number;
}
