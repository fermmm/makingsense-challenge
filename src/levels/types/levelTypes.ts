export enum Tile {
   Walkable,
   Obstacle,
}

export interface Level {
   startPos: Vector;
   finishPos: Vector;
   map: Tile[][];
}

export interface Vector {
   x: number,
   y: number
}