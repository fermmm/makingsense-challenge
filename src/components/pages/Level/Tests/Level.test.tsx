import { mount } from 'enzyme';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { LevelData, TileData } from '../../../../levels/tools/types/levelTypes';
import Level from '../Level';

describe('<Level />', () => {
   const testLevel: LevelData = {
      startPos: { x: 0, y: 0 },
      finishPos: { x: 2, y: 1 },
      movesAvailable: 3,
      map: [
         [TileData.Walkable, TileData.Walkable, TileData.Obstacle],
         [TileData.Walkable, TileData.Walkable, TileData.Walkable],
      ],
   };
   const onWin = jest.fn();
   const onLose = jest.fn();

   test('Renders correctly', () => {
      const component = mount(<Level levelData={testLevel} onWin={onWin} onLoose={onLose} />);
      expect(component.html()).toMatchSnapshot();
      component.unmount();
   });

   test('Winning works correctly', () => {
      const component = mount(<Level levelData={testLevel} onWin={onWin} onLoose={onLose} />);

      act(() => {
         window.dispatchEvent(new KeyboardEvent('keydown', { code: 'ArrowRight' }));
      });
      act(() => {
         window.dispatchEvent(new KeyboardEvent('keydown', { code: 'ArrowDown' }));
      });
      act(() => {
         window.dispatchEvent(new KeyboardEvent('keydown', { code: 'ArrowRight' }));
      });
      expect(onWin).toHaveBeenCalledTimes(1);

      expect(component.html()).toMatchSnapshot();
      component.unmount();
   });

   test('Loosing works correctly', () => {
      const component = mount(<Level levelData={testLevel} onWin={onWin} onLoose={onLose} />);

      act(() => {
         window.dispatchEvent(new KeyboardEvent('keydown', { code: 'ArrowRight' }));
      });
      act(() => {
         window.dispatchEvent(new KeyboardEvent('keydown', { code: 'ArrowDown' }));
      });
      act(() => {
         window.dispatchEvent(new KeyboardEvent('keydown', { code: 'ArrowLeft' }));
      });
      expect(onLose).toHaveBeenCalledTimes(1);

      expect(component.html()).toMatchSnapshot();
      component.unmount();
   });
});
