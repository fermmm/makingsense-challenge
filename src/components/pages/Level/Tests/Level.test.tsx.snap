// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Level /> Loosing works correctly 1`] = `"<div class=\\"mainContainer\\"><div class=\\"mapContainer\\"><div class=\\"column undefined\\"><div class=\\"row undefined\\"><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div><div class=\\"tile\\" style=\\"background-color: rgb(173, 173, 173); width: 80px; height: 80px;\\"></div></div><div class=\\"row undefined\\"><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div></div></div><div class=\\"mapEntity undefined\\" style=\\"width: 80px; height: 80px; left: 160px; top: 80px;\\"><img class=\\"finishTile\\" src=\\"\\"></div><div class=\\"mapEntity undefined\\" style=\\"width: 80px; height: 80px; left: 0px; top: 80px;\\"><img class=\\"player\\" src=\\"\\"></div></div>Moves left: 0</div>"`;

exports[`<Level /> Renders correctly 1`] = `"<div class=\\"mainContainer\\"><div class=\\"mapContainer\\"><div class=\\"column undefined\\"><div class=\\"row undefined\\"><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div><div class=\\"tile\\" style=\\"background-color: rgb(173, 173, 173); width: 80px; height: 80px;\\"></div></div><div class=\\"row undefined\\"><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div></div></div><div class=\\"mapEntity undefined\\" style=\\"width: 80px; height: 80px; left: 160px; top: 80px;\\"><img class=\\"finishTile\\" src=\\"\\"></div><div class=\\"mapEntity undefined\\" style=\\"width: 80px; height: 80px; left: 0px; top: 0px;\\"><img class=\\"player\\" src=\\"\\"></div></div>Moves left: 3</div>"`;

exports[`<Level /> Winning works correctly 1`] = `"<div class=\\"mainContainer\\"><div class=\\"mapContainer\\"><div class=\\"column undefined\\"><div class=\\"row undefined\\"><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div><div class=\\"tile\\" style=\\"background-color: rgb(173, 173, 173); width: 80px; height: 80px;\\"></div></div><div class=\\"row undefined\\"><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div><div class=\\"tile\\" style=\\"background-color: white; width: 80px; height: 80px;\\"></div></div></div><div class=\\"mapEntity undefined\\" style=\\"width: 80px; height: 80px; left: 160px; top: 80px;\\"><img class=\\"finishTile\\" src=\\"\\"></div><div class=\\"mapEntity undefined\\" style=\\"width: 80px; height: 80px; left: 160px; top: 80px;\\"><img class=\\"player\\" src=\\"\\"></div></div>Moves left: 0</div>"`;
